nav2-usability/scenario_code/lib/nested-routing/nested_routing_go_router_prototype.dart

```diff
@@ -0,0 +1,217 @@
+// Copyright 2022, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:go_router_prototype/go_router_prototype.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class BooksApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routeInformationParser: _router.parser,
+      routerDelegate: _router.delegate,
+    );
+  }
+
+  late final _router = GoRouter(
+    routes: [
+      ShellRoute(
+        path: '/',
+        redirect: (routeState) async {
+          if (routeState.path == '/') {
+            return '/books/new';
+          }
+        },
+        builder: (context, child) {
+          late final int selectedIndex;
+          final childPath = RouteState.of(context).activeChild?.path;
+          if (childPath == null || childPath == 'books') {
+            selectedIndex = 0;
+          } else if (childPath == 'settings') {
+            selectedIndex = 1;
+          }
+
+          return AppScreen(
+            currentIndex: selectedIndex,
+            child: child,
+          );
+        },
+        routes: [
+          ShellRoute(
+            path: 'books',
+            redirect: (routeState) async {
+              if (routeState.path == '/books') {
+                return '/books/new';
+              }
+            },
```

Comment:
What is the value of these redirects? I assume the `/` is necessary when the app is first opened, but what's this one for? Would you get an error without this redirect? Is it possible for a user to hit this redirect without some logical bug in the app?

The reason I ask is because this looks like noisy boilerplate. It'd be nice if we could remove this redirect!

Reply:
This should be using `ShellRoute`'s `defaultRoute API. In this scenario, it's important to synchronize the sub-route ("/new" or "/all") with the state of the TabBar, and set the initial value.

Reply:
https://github.com/flutter/uxr/pull/82/commits/ace3cd95b10eb33ff41565a9bb03bd858965091d